### PR TITLE
fix: move ci/cd to github actions

### PR DESCRIPTION
**Pull request by jamesxu123**
### Tickets:

-   HAC-32
### List of changes:

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Move CI/CD to GitHub Actions due to deprecation of travis-ci.org

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] New release
-   [ ] This change requires a documentation update

### How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

-   [ ] Test A
-   [ ] Test B

**Test Configuration**:

**Firmware version:**
**Hardware:**
**Toolchain:**
**SDK:**

### Checklist:

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings
-   [ ] Listed change(s) in the Changelog
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] I have made corresponding changes to the documentation
-   [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ dev ]
```
Comment (pierreTklein) [inline .github/workflows/node.js.yml:10]: How about `master` branch? In case someone pushes to that branch in a  rush right before the hackathon.

Comment (jamesxu123) [reply .github/workflows/node.js.yml:10]: Thanks for that. I forgot about `master`!

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      mongodb:
+        image: mongo
+        ports:
+          - 27017:27017
+    environment: testing
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
```
Comment (krubenok) [inline .github/workflows/node.js.yml:24]: This doesn't match the node version specified in `.nvmrc` If moving to 14.x is intentional, `.nvmrc` should also be moved.

File: .idea/vcs.xml [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+        <option value="dev" />
+      </list>
+    </option>
+  </component>
   <component name="VcsDirectoryMappings">
     <mapping directory="$PROJECT_DIR$" vcs="Git" />
   </component>
-</project>
\ No newline at end of file
+</project>
```
Comment (krubenok) [inline .idea/vcs.xml]: Does anyone still use Jetbrains IDEs? Can this whole .idea folder be deleted?

**Review by krubenok — COMMENTED** [review]
Overall, looks fine, but I think it'd be good to either merge #691 first or merge that branch into this so that we can have a full view of the impact on CI/CD before calling it good. 

Also: `travis.yml` should be deleted as part of this PR. 

Edit: this would also close #660

File: .idea/vcs.xml (hunk 3 above)
Comment (jamesxu123) [reply .idea/vcs.xml]: I use PhpStorm, let's keep it for now.

**Comment by jamesxu123** [comment]
> Overall, looks fine, but I think it'd be good to either merge #691 first or merge that branch into this so that we can have a full view of the impact on CI/CD before calling it good.
> 
> Also: `travis.yml` should be deleted as part of this PR.
> 
> Edit: this would also close #660

#691 is still in progress and it's estimated to take quite a long time because a lot of manual work is involved. Since Travis is deprecated, I believe it's best to get something functional in place so our other PRs won't be blocked. Good catch on the deletion of travis.yml though thanks!

**Review by krubenok — APPROVED** [review]
🚢 it! 


Maybe time to investigate re-enabling dependabot now to start bailing out of dependency hell

**Pull request merged**